src/UI/Seller/src/app/sellers/components/locations/seller-location-edit/seller-location-edit.component.ts

```diff
@@ -0,0 +1,186 @@
+  set location(sellerLocation: Address) {
+    if (sellerLocation.ID) {
+      this.handleSelectedLocationChange(sellerLocation)
+    } else {
+      this.handleSelectedLocationChange(
+        this.sellerLocationService.emptyResource
+      )
+    }
+  }
+  @Input()
+  filterConfig
+  @Input()
+  suggestedAddresses: ListPage<Address>
+  @Output()
+  updateResource = new EventEmitter<any>()
+  @Output()
+  selectAddress = new EventEmitter<any>()
+  @Output()
+  canDelete = new EventEmitter<boolean>()
+
+  constructor(
+    private middleware: MiddlewareAPIService,
+    private activatedRoute: ActivatedRoute,
+    private sellerLocationService: SellerAddressService
+  ) {
+    this.countryOptions = GeographyConfig.getCountries()
+  }
+
+  ngOnChanges() {
+    this.setLocationID()
+  }
+
+  async handleSelectedLocationChange(location: Address): Promise<void> {
+    // Bug with the SDK where GetRateList has wrong return type
+    try {
+      HeadStartSDK.ExchangeRates.GetRateList().then((res) => {
+        this.currencyOptions = res.Items
+        location.Country && this.setFlag(location.Country)
+      })
+    } finally {
+      this.createSellerLocationForm(location)
+    }
+  }
+
+  createSellerLocationForm(sellerLocation: Address) {
+    this.resourceForm = new FormGroup({
+      AddressName: new FormControl(
+        sellerLocation.AddressName,
+        Validators.required
+      ),
+      CompanyName: new FormControl(
+        sellerLocation.CompanyName,
+        Validators.required
+      ),
+      Street1: new FormControl(sellerLocation.Street1, Validators.required),
+      Street2: new FormControl(sellerLocation.Street2),
+      City: new FormControl(sellerLocation.City, Validators.required),
+      State: new FormControl(sellerLocation.State, Validators.required),
+      Zip: new FormControl({ value: sellerLocation.Zip, disabled: true }, [
+        Validators.required,
+        ValidateUSZip || ValidateCAZip,
+      ]),
+      Country: new FormControl(sellerLocation.Country, Validators.required),
+      Phone: new FormControl(sellerLocation.Phone, ValidatePhone),
+    })
+    this.setZipValidator()
+  }
+
+  setFlag(locationCountry: string) {
+    const currency = this.getCurrencyFromCode(locationCountry)
+    this.flag = this.getFlagForCountry(currency)
+  }
+
+  setZipValidator() {
+    const zipControl = this.resourceForm.get('Zip')
+    this.countryHasBeenSelected =
+      this.resourceForm.controls['Country'].value !== ''
+    this.resourceForm
+      .get('Country')
+      .valueChanges.pipe(takeWhile(() => this.alive))
+      .subscribe((code) => {
+        const currency = this.getCurrencyFromCode(code)
+        this.flag = this.getFlagForCountry(currency)
+        this.countryHasBeenSelected = code !== ''
+        if (code !== null) zipControl.enable()
+        if (code === 'CA') {
+          zipControl.setValidators(ValidateCAZip)
+        } else {
+          zipControl.setValidators(ValidateUSZip)
+        }
+      })
+  }
```

Comment:
Angular has a built-in notion of validators, can you use that instead? The benefit is that we can centralize the logic and use it in other parts of the application that might need that same validation. It should live in src/app/shared/validators

Here's a guide I found, I'm sure there are others. https://indepth.dev/posts/1319/the-best-way-to-implement-custom-validators